src/AnalyzerPowerPack/Core/Design/CA1052DiagnosticAnalyzer.cs

```diff
@@ -100,13 +104,18 @@ internal static class CA1052Extensions
         /// "qualifying member" (<see cref="IsQualifyingMember(ISymbol)"/>) and no
         /// "disqualifying members" (<see cref="IsDisqualifyingMember(ISymbol)"/>).
         /// </remarks>
-        internal static bool IsStaticHolderType(this INamedTypeSymbol symbol)
+        internal static bool IsStaticHolderType(this INamedTypeSymbol symbol, Compilation compilation)
         {
             if (symbol.TypeKind != TypeKind.Class)
             {
                 return false;
             }
 
+            if (symbol.BaseType != compilation.GetTypeByMetadataName("System.Object"))
```

Comment:
Are there any incomplete code scenarios? for example - a missing base list 'class foo :` - if we still have a base type there, you can leave out the null check.